staging/src/k8s.io/kubectl/pkg/util/slice/slice_test.go

```diff
@@ -29,3 +30,44 @@ func TestSortInts64(t *testing.T) {
 		t.Errorf("func Ints64 didnt sort correctly, %v !- %v", src, expected)
 	}
 }
+
+func TestToSet(t *testing.T) {
```

Comment:
Love the unit tests! Would it be possible to add a regression test for the bug itself as well?

Reply:
I will write a unit test for that.

By the way, the existing prune unit test ([`TestApplyPruneObjects`](https://github.com/kubernetes/kubernetes/blob/cddbb0c56397448ac0489f0473a26601c1feece8/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go#L663)) actually doesn't test prune at all.  If you remove the line [`cmd.Flags().Set("prune", "true")`](https://github.com/kubernetes/kubernetes/blob/cddbb0c56397448ac0489f0473a26601c1feece8/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go#L697), the test still passes! :open_mouth:

Reply:
I added some unit tests for prune.

The test case that prevents a regression is `prune with allowlist specifying the same resource type multiple times should not fail`